proxmox/resource_vm_qemu.go

```diff
@@ -1952,26 +1953,36 @@ func initConnInfo(
 	}
 	// todo - log a warning if we couldn't get an IP
 
-	d.Set("default_ipv4_address", sshHost)
-
 	if config.HasCloudInit() {
 		log.Print("[DEBUG][initConnInfo] vm has a cloud-init configuration")
-		if sshHost == "" {
-			log.Print("[DEBUG][initConnInfo] not found an ip configuration yet")
-			_, ipconfig0Set := d.GetOk("ipconfig0")
-			if ipconfig0Set {
-				vmState, err := client.GetVmState(vmr)
-				if err != nil {
-					return err
-				}
+		_, ipconfig0Set := d.GetOk("ipconfig0")
+		if ipconfig0Set {
+			vmState, err := client.GetVmState(vmr)
+			if err != nil {
+				return err
+			}
 
-				if d.Get("ipconfig0").(string) != "ip=dhcp" || vmState["agent"] == nil || vmState["agent"].(float64) != 1 {
-					// parse IP address out of ipconfig0
-					ipMatch := rxIPconfig.FindStringSubmatch(d.Get("ipconfig0").(string))
+			if d.Get("ipconfig0").(string) != "ip=dhcp" || vmState["agent"] == nil || vmState["agent"].(float64) != 1 {
+				// parse IP address out of ipconfig0
+				ipMatch := rxIPconfig.FindStringSubmatch(d.Get("ipconfig0").(string))
+				if sshHost == "" {
 					sshHost = ipMatch[1]
 				}
+				ipconfig0 := net.ParseIP(strings.Split(ipMatch[1], ":")[0])
+				for _, iface := range interfaces {
```

Comment:
as per the previous comment, with interfaces = nill this will loop until the end of time

Reply:
Copy the check for errInterface != nil
Or should i ruse the err variable at this place?

Reply:
look at this schenario(like mine)
I make a vm with cloud init, the first part of the function will find the ip in some minutes and its' ok, but in my cloudinit conf i make the vm reboot after the install of qemu-guest-agent(mainly to reboot to the updated kernel after the system upgrade for example)
the ip should be ok but then with your PR we check it another time just to overwrite it with what is in the cloudinit conf
but in this situation the vm is rebooting so the check for the interfaces give us an error 500 and the code crashes because it's not handled
this will not happen 100% because it depends on how fast the process is, but with a lot of vm created at once(like a lot of people there like to do thinking they have AWS grade servers), a lot of random things can happen

Reply:
Ok, now i unterstand. 

So i moved interface to a "function variable" and in the "default" IP loop i rewrite the variable interface after this loop i just reuse it in the cloudinit entry. So when the first loop won't end in an infinity loop then the second can't end in one. 

Should i search for a other solution?